### PR TITLE
WasmFS JS API: Implement mknod and create

### PR DESCRIPTION
**Pull request by jameshu15869**
This PR implements both `mknod` and `create`, since `create` seems to just call `mknod` in the original JS API. 

As a side note, I noticed that `mkdev` is closely related to `mknod`. However, the Emscripten documentation gives examples of using methods like `registerDevice` before calling `mkdev`. I tried looking through `library_wasmfs.js` and `syscalls.cpp` to find code for devices, but didn't see anything obvious. Where should I look for WasmFS devices?

### CODE REVIEW

**Comment by kripken** [comment]
About `mkdev`, I don't think we have support for devices in WasmFS atm, but I'm not sure. @tlively did we have a plan for them, or a decision to not support them?

File: src/library_wasmfs.js [hunk 1]
```diff
@@ -126,6 +126,12 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       })
     },
     // TODO: create
```
Comment (kripken) [inline src/library_wasmfs.js:128]:
```suggestion
```
The TODO is now TODONE :smile:

File: src/library_wasmfs.js [hunk 2]
```diff
@@ -126,6 +126,12 @@ FS.createPreloadedFile = FS_createPreloadedFile;
...
+    create: (path, mode) => {
+      mode = mode !== undefined ? mode : 438 /* 0666 */;
+      mode &= {{{ cDefs.S_IALLUGO }}};
+      mode |= {{{ cDefs.S_IFREG }}};
```
Comment (kripken) [inline src/library_wasmfs.js:132]: Were these settings copied from the JS FS, or do they stem from the docs, or otherwise? A comment would be good here.

**Review by kripken — COMMENTED** [review]
Looks great aside from minor comments.

File: src/library_wasmfs.js (hunk 2 above)
Comment (jameshu15869) [reply src/library_wasmfs.js:132]: The settings were copied from the JS FS API and I've added a comment accordingly.

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
(lgtm aside from conflicts)

**Comment by tlively** [comment]
> About `mkdev`, I don't think we have support for devices in WasmFS atm, but I'm not sure. @tlively did we have a plan for them, or a decision to not support them?

I don't think we had a concrete plan, but I don't think we had a decision to omit them, either. Broadly, it would be possible to implement the `mkdev` API by having it create new WasmFS backends that operate in terms of the JS stream functions provided by the user. Since WasmFS doesn't have the same stream objects, the functions might need to be different, though.

This might be an interesting avenue for exploring how to make the legacy backends like IDBFS available under WasmFS, since it seems like a similar problem.

Skipping `mkdev` for now seems like the best move.

**Pull request merged**